### PR TITLE
Fix no rows present not displaying correctly

### PR DESCRIPTION
**Pull request by matttdawson**
Fixed no rows overlay to be an actual component, otherwise it does not update with data changes.

### CODE REVIEW

File: src/components/Grid.tsx [hunk 1]
```diff
@@ -215,7 +215,11 @@ export const Grid = ({ rowSelection = "multiple", "data-testid": dataTestId, ...
 
   const noRowsOverlayComponent = useCallback(
     (event: AgGridEvent) => {
-      const hasData = (params.rowData?.length ?? 0) > 0;
+      let hasData = false;
+      // This is the only way to get a correct data rows exists flag.  params.rowData.length doesn't work here.
+      event.api.forEachNode(() => {
```
Comment (jfee-linz) [inline src/components/Grid.tsx:220]: Does this work?

```
event.api.getModel().getRowCount()
```

**Review by jfee-linz — CHANGES_REQUESTED** [review]

File: src/components/Grid.tsx (hunk 1 above)
Comment (matttdawson) [reply src/components/Grid.tsx:220]: No.  I'm still looking in to this. I don't like the solution, but for some reason the rowData just isn't updating in the useEffect.

Comment (matttdawson) [reply src/components/Grid.tsx:220]: Even though filtered rows has updated, and the forEach as well.

Comment (matttdawson) [reply src/components/Grid.tsx]: I figured it out.  Of course react doesn't treat functions as components, I just had to change it to a real component.

File: src/components/Grid.tsx [hunk 2]
```diff
@@ -213,21 +214,9 @@ export const Grid = ({ rowSelection = "multiple", "data-testid": dataTestId, ...
...
-          {!hasData
-            ? params.noRowsOverlayText ?? "There are currently no rows"
-            : !hasFilteredData
-            ? "All rows have been filtered"
-            : ""}
-        </span>
-      );
-    },
-    [params.noRowsOverlayText, params.rowData?.length],
+  const noRowsOverlayComponent = useMemo(
```
Comment (jfee-linz) [inline src/components/Grid.tsx:217]: Can you eliminate the memo entirely?  Also, can the content be `JSX.Element` for custom icons/etc?

```
noRowsOverlayComponent={<NoRowsOverlayComponent>{props.noRowsContent}</NoRowsOverlayComponent>}
```

Comment (matttdawson) [reply src/components/Grid.tsx]: Fixed.

**Review by jfee-linz — APPROVED** [review]
so main changes now are renaming the component and adding storybook?

[edit] was looking at a subset of changes.  This looks good :)

**Pull request merged**